quinn-proto/src/token.rs

```diff
@@ -71,6 +72,22 @@ impl RetryToken {
             issued,
         })
     }
+
+    /// Ensure that this token validates an `Incoming`, and construct its token state
+    fn validate(
```

Comment:
Djc, I know you've been [interested](https://github.com/quinn-rs/quinn/pull/1912#discussion_r1886597384) in in-lining this into `validate`. However, the place in `validate` where this is called is within an `and_then` closure, and you [didn't](https://github.com/quinn-rs/quinn/pull/1912#discussion_r1856334781) really like of having a big amount of logic in-lined into an `and_then` either. On that matter, going through with NEW_TOKEN usage would likely add even more logic here--although of course I can do it one way for this PR and then change it back in the future.

In any case, I think that having this method be separate makes the code easier to understand, because it very naturally reifies how I mentally compartmentalize different concerns in processing tokens.

Still, though, I can inline it if wanted.

Reply:
Thanks for flagging this, going to merge this as-is and see if there's further refactoring that makes sense to me.